cmd/scollector/main.go

```diff
@@ -245,17 +245,27 @@ func main() {
 		}
 		collect.MaxQueueLen = conf.MaxQueueLen
 	}
-	maxMemMegaBytes := uint64(500)
+	maxMemMB := uint64(500)
 	if conf.MaxMem != 0 {
-		maxMemMegaBytes = conf.MaxMem
+		maxMemMB = conf.MaxMem
 	}
 	go func() {
-		maxMemBytes := maxMemMegaBytes * 1024 * 1024
 		var m runtime.MemStats
 		for range time.Tick(time.Second * 30) {
 			runtime.ReadMemStats(&m)
-			if m.Alloc > maxMemBytes {
-				panic(fmt.Sprintf("memory max reached: (current: %v bytes, max: %v bytes)", m.Alloc, maxMemBytes))
+			allocMB := m.Alloc / 1024 / 1024
+			if allocMB > maxMemMB {
+				panic(fmt.Sprintf("memory max runtime reached: (current alloc: %v megabytes, max: %v megabytes)", allocMB, maxMemMB))
+			}
+			//See proccess_windows.go and process_linux.go for total process memory usage.
```

Comment:
Is it true that this will only work if they happen to be monitoring the scollector process? Or maybe just any process monitoring enabled at all?